File: problems/2256_minimum_average_difference/solution.py
from typing import List


class Solution:
    def minimumAverageDifference(self, nums: List[int]) -> int:
        n = len(nums)
        total = sum(nums)
        left_sum = 0
        min_diff = float('inf')
        result = 0
        for i in range(n):
            left_sum += nums[i]
            left_avg = left_sum // (i + 1)
            right_avg = (total - left_sum) // (n - i - 1) if i < n - 1 else 0
            diff = abs(left_avg - right_avg)
            if diff < min_diff:
                min_diff = diff
                result = i
        return result

File: problems/2256_minimum_average_difference/Solution.java
class Solution {
    public int minimumAverageDifference(int[] nums) {
        int n = nums.length;
        long total = 0;
        for (int x : nums) total += x;
        long leftSum = 0;
        long minDiff = Long.MAX_VALUE;
        int result = 0;
        for (int i = 0; i < n; i++) {
            leftSum += nums[i];
            long leftAvg = leftSum / (i + 1);
            long rightAvg = i < n - 1 ? (total - leftSum) / (n - i - 1) : 0;
            long diff = Math.abs(leftAvg - rightAvg);
            if (diff < minDiff) { minDiff = diff; result = i; }
        }
        return result;
    }
}
